rdr_service/api/nph_intake_api.py

```diff
@@ -0,0 +1,216 @@
+
+            if entry['resource'].get('class'):
+                activity_name = activity_source = entry['resource']['class']['code']
+
+                if 'module' in activity_name:
+                    activity_name = 'enrollment'
+
+            if entry['resource'].get('serviceType'):
+                activity_source = entry['resource']['serviceType']['coding'][0]['code']
+
+            activity_name = activity_name.lower()
+            current_activity = list(filter(lambda x: x.name.lower() == activity_name,
+                                           self.current_activities))
+            if not current_activity:
+                raise BadRequest(f'Cannot reconcile activity type bundle_id: {self.bundle_identifier}')
+
+            return ActivityData(
+                id=current_activity[0].id,
+                name=activity_name,
+                source=activity_source
+            )
+
+        except KeyError as e:
+            return BadRequest(f'Key error on activity lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    def get_site_id(self, entry: dict):
+        try:
+            pairing_site_code = entry['resource']['type'][0]['coding'][0]['code']
+
+            if not pairing_site_code:
+                raise BadRequest(f'Cannot find site pairing code: bundle_id: {self.bundle_identifier}')
+
+            site = self.nph_site_dao.get_site_id_from_external(external_id=pairing_site_code)
+
+            if not site:
+                raise BadRequest(f'Cannot find site from site code: bundle_id: {self.bundle_identifier}')
+
+            return site.id
+
+        except KeyError as e:
+            raise BadRequest(f'Key error on site lookup: {e} bundle_id: {self.bundle_identifier}')
+
+    def get_event_type_id(self, *, activity_name, activity_source):
+        event_type_dao_instance_items = {k: v for k, v in self.__dict__.items() if 'type_dao' in k}
+
+        if activity_name not in event_type_dao_instance_items.keys():
+            return 1
+
+        event_type_dao = event_type_dao_instance_items[f'nph_{activity_name}_type_dao']
+        event_activity = event_type_dao.get_event_by_source_name(source_name=activity_source)
+
+        if not event_activity:
+            raise BadRequest(f'Cannot find event type: bundle_id: {self.bundle_identifier}')
+
+        return event_activity.id
+
+    def build_event_dao_map(self):
+        event_dao_map = {}
+        event_dao_instance_items = {k: v for k, v in self.__dict__.items() if 'event_dao' in k}
+        for activity in self.current_activities:
+            event_dao_map[f'{activity.name.lower()}'] = event_dao_instance_items[
+                f'nph_{activity.name.lower()}_event_dao']
+        return event_dao_map
+
+    @classmethod
+    def extract_participant_id(cls, participant_obj: dict) -> str:
+        participant_id = participant_obj['resource']['identifier'][0]['value']
+        participant_id = participant_id.split('/')[-1]
```

Comment:
The participant IDs we receive have a `1000` appended to the front of the ID the same way in AoU we have the `P` character. Can you add a slice here to remove the 1000?

Reply:
Done 
https://github.com/all-of-us/raw-data-repository/pull/3344/commits/7ed37ffeb96955d88dd24ae291c0820bca9b36d4